### PR TITLE
Python3 fix return item handle

### PR DESCRIPTION
**Pull request by devang-dreambits**
[FIX] updated method getHandle which was defined wrong; still getting 400 bad request error

### CODE REVIEW

File: bol/plaza/api.py [hunk 1]
```diff
@@ -378,15 +387,42 @@ def __init__(self, api):
...
-        xml = self.request('PUT', '/{}/handle'.format(orderId), params={
-            'StatusReason': status_reason,
-            'QuantityReturned': qty_return
-        })
-        return ProcessStatus.parse(self.api, xml)
+        try:
+            xml = self.request('GET', path="/unhandled", accept="application/xml")
+            return ReturnItems.parse(self.api, xml)
+        except:
+            print("Got into Exception \n")
```
Comment (karan-dreambits) [inline bol/plaza/api.py]: Any special need to put exceptions here ?

File: bol/plaza/api.py [hunk 2]
```diff
@@ -136,6 +136,15 @@ def create_request_inbound_xml(self, root, **kwargs):
 <{root} xmlns="https://plazaapi.bol.com/services/xsd/v1/plazaapi.xsd">
 {elements}
 </{root}>
+""".format(root=root, elements=elements)
+        return xml
+
+    def create_request_return_items_handle_xml(self, root, **kwargs):
```
Comment (karan-dreambits) [inline bol/plaza/api.py]: what difference does this function have from the function `create_request_xml` ?

File: bol/plaza/api.py [hunk 3]
```diff
@@ -378,15 +387,42 @@ def __init__(self, api):
...
+                version=self.api.version,
+                path='/{}/handle'.format(returnNumber))
+            response = self.api.request('PUT', uri, params=params,
+                                        data=xml, accept="application/xml")
+            return ProcessStatus.parse(self.api, response)
+        except:
+            print("Got into Exception \n")
+            traceback.print_exc()
+
+    def getHandle(self, returnNumber, status_reason, qty_return):
```
Comment (karan-dreambits) [inline bol/plaza/api.py]: This function needs to removed

File: bol/plaza/api.py [hunk 4]
```diff
@@ -378,15 +387,42 @@ def __init__(self, api):
...
-        })
-        return ProcessStatus.parse(self.api, xml)
+        try:
+            xml = self.request('GET', path="/unhandled", accept="application/xml")
+            return ReturnItems.parse(self.api, xml)
+        except:
+            print("Got into Exception \n")
+            traceback.print_exc()
+
+    def getHandle_new(self, returnNumber, status_reason, qty_return, params=None):
```
Comment (karan-dreambits) [inline bol/plaza/api.py]: why put "_new" in function name ?
Anyways we are not "getting" unhandled request, we are handling the return item request, 
so name it as :

handleReturnItem

Comment (devang-dreambits) [reply bol/plaza/api.py]: I'll rename it.

File: bol/plaza/api.py (hunk 3 above)
Comment (devang-dreambits) [reply bol/plaza/api.py]: Ok.

File: bol/plaza/api.py (hunk 2 above)
Comment (devang-dreambits) [reply bol/plaza/api.py]: There is no difference, I'll remove the redundant method.

File: bol/plaza/api.py (hunk 1 above)
Comment (devang-dreambits) [reply bol/plaza/api.py]: No specific one, but I had added earlier as part of testing to see if any errors are coming.

File: tests/test_plaza.py [hunk 5]
```diff
@@ -750,3 +750,37 @@ def unhandled_return_items_stub(url, request):
...
+   <ns1:eventType>HANDLE_RETURN_ITEM</ns1:eventType>
+   <ns1:description>Handle the return item with returnNumber 65380525</ns1:description>
+   <ns1:status>PENDING</ns1:status>
+   <ns1:createTimestamp>2019-02-19T09:08:36.629+01:00</ns1:createTimestamp>
+   <ns1:Links>
+      <ns1:link ns1:rel="self" ns1:href="https://plazaapi.bol.com/services/rest/process-status/v2/112748417" ns1:method="GET" />
+   </ns1:Links>
+</ns1:ProcessStatus>"""
+
+HANDLED_RETUERN_ITEMS_RESPONSE = [RI_handle_str_1]
```
Comment (karan-dreambits) [inline tests/test_plaza.py]: Typo

File: bol/plaza/api.py [hunk 6]
```diff
@@ -378,15 +378,30 @@ def __init__(self, api):
...
+                )
+            uri = '/services/rest/{group}/{version}{path}'.format(
+                group=self.group,
+                version=self.api.version,
+                path='/{}/handle'.format(returnNumber))
+            response = self.api.request('PUT', uri, params=params,
+                                        data=xml, accept="application/xml")
+            return ProcessStatus.parse(self.api, response)
+        except:
+            print("Got into Exception \n")
```
Comment (karan-dreambits) [inline bol/plaza/api.py]: Either we need to add exceptions everywhere or remove everywhere.
And having general exception catching is very bad practice.
Can we remove it if you don't have any specific exception catching needed here ?

File: bol/plaza/api.py [hunk 7]
```diff
@@ -378,15 +378,30 @@ def __init__(self, api):
...
-        xml = self.request('PUT', '/{}/handle'.format(orderId), params={
-            'StatusReason': status_reason,
-            'QuantityReturned': qty_return
-        })
-        return ProcessStatus.parse(self.api, xml)
+        try:
+            xml = self.request('GET', path="/unhandled", accept="application/xml")
+            return ReturnItems.parse(self.api, xml)
+        except:
+            print("Got into Exception \n")
```
Comment (karan-dreambits) [inline bol/plaza/api.py]: Same as before, why keep general exception catching ?

Comment (devang-dreambits) [reply bol/plaza/api.py]: I forgot to remove it.

**Pull request merged**